openstack/tools/charmed_openstack_functest_runner.sh

```diff
@@ -133,14 +133,20 @@ FIP_MIN_ABC=${FIP_MIN%.*}
 FIP_MIN_D=${FIP_MIN##*.}
 FIP_MIN=${FIP_MIN_ABC}.$(($FIP_MIN_D + 64))
 
-CIDR_OAM=$(openstack subnet show subnet_${OS_USERNAME}-psd -c cidr -f value)
-OAM_MAX=$(ipcalc $CIDR_OAM| awk '$1=="HostMax:" {print $2}')
-OAM_MIN=$(ipcalc $CIDR_OAM| awk '$1=="HostMin:" {print $2}')
-OAM_MIN_ABC=${OAM_MIN%.*}
-OAM_MAX_D=${OAM_MAX##*.}
-# Picking last two addresses and hoping they dont get used by Neutron.
-export {OS,TEST}_VIP00=${OAM_MIN_ABC}.$(($OAM_MAX_D - 1))
-export {OS,TEST}_VIP01=${OAM_MIN_ABC}.$(($OAM_MAX_D - 2))
+# Setup VIPs
+NUM_VIPS=2
+for ((i=$NUM_VIPS;i;i-=1)); do
+    # We use the same naming convention as ../tools/allocate_vips.sh to avoid conflicts and re-use
```

Comment:
The logic here is fine and I'm fine with it, but previously I had suggested using what already exists instead. This logic is creating 2 VIPs, the tools/allocate_vips.sh script creates 20. I am not sure if the tools/allocate_vips.sh script will fail if 2 are already allocated. Sounds easier to me if you just do a openstack port list | grep ps6-vip and if there is any result then you skip, otherwise you run tools/allocate_vips.sh to allocate 20.

Reply:
I forgot to ask in the above comment: had you considered that approach I suggested? I'm thinking maybe you did and then you implemented as you are proposing for a specific reason.

Reply:
actually my point was more that the allocatte_vips script is creating with specific addresses which as brian pointed out is not necessary and prone to conflicts at create time but fixing it will be too much work for this patchset so I will followup

Reply:
good point, ideally everyone should run allocate_vips before they have any running resources, but if they don't then it would cause problems.